### ISSUE TITLE
External subtitles not loaded when "Play next video automatically" enabled.

### ISSUE THREAD

**Issue by sergiopistone** [xbmc/xbmc#14947]
### Describe the bug
Here is a clear and concise description of what the problem is:

I've created a media source for TV Shows pointing to a local file system folder (probably not relevant but, just in case, the information provider is set to "Local information only" as I use Ember Media Manager for collecting shows info).
When playing the files through the Media Library node AND "Play next video automatically" is enabled for "Episodes", the external subtitles files are not displayed nor loaded (the list of available subtitles while playing the video is empy). The subtitle files are named appropriately though (using the same name as the video, but with a language suffix and the subtitle type extension), and they are loaded and displayed properly when the same files are played through the media source node directly (instead of "TV Shows", use Videos -> Files -> %Media Source Name%) or even when played through the Media Library "TV Shows" node but with play next episodes disabled.

The problem might be related to this other issue: https://github.com/xbmc/xbmc/issues/14841

The following forum thread mentions the problem described (and also provides a work-around to the issue): https://forum.kodi.tv/showthread.php?tid=326482

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The external subtitles files should be loaded and displayed, in the same way they are when files are played through through the files nodes or when play next episodes automatically is disabled.

## Actual Behavior

As stated above, the external subtitles files are not loaded when playing the files through the Media Library node if "Play next video automatically" is enabled for "Episodes". 

## Possible Fix

This seems to be related to the different way Kodi addresses the files when playing through the Media Library node vs when playing through the Files node (see attached screenshots).

### To Reproduce

1. Create a TV Shows media source pointing to a local folder with shows containing external subtitle files.
2. Enable "Play next video automatically" for Episodes by going to Settings -> Player -> Video
3. Go to TV Shows and attempt to play an episode of one of the added shows. Subtitles will not be displayed, attempting to browse the available subtitles won't show them either.

### Debuglog

The debuglog can be found here:
[kodi.log](https://github.com/xbmc/xbmc/files/2614222/kodi.log)

### Screenshots 
Here are some links or screenshots to help explain the problem:

Browse subtitles, when playing file through TV Shows node with play next episode automatically enabled.
![image](https://user-images.githubusercontent.com/40037617/48994943-717e0100-f124-11e8-9ff8-4f26691a3951.png)
Browse subtitles, when playing file through created media source node or through TV Shows node with play next episode automatically disabled.
![image](https://user-images.githubusercontent.com/40037617/48994969-a25e3600-f124-11e8-82bc-90ff0b34a044.png)

(Notice the URL the subtitles are being searched on)

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberri-Pi
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 10
 - Kodi version: Kodi 18 (Leia) RC1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by kleczerx** [comment]
I also experience no subtitle operation when "Play next video automatically" is activated. This occurs when selecting a video from the face of the main window or via the main side-menu in "Movies" and "TV Shows". This fault existed for all Beta, Alpha and the recent Release Candidate 1. If automatic play is active, THEN normal operation does occur if the video file played is selected via "Videos" off the main side-menu. My play vehicle for KODI is in both Windows 10 OS and Nvidea Shield media player.

**Comment by Blakeyjr** [comment]
This is the same issue: https://github.com/xbmc/xbmc/issues/14841

**Comment by skonvols2k** [comment]
Hi,
RC3 this problem still occurs. Please fix! Thanks

**Comment by pkerling** [comment]
@FernetMenta Only a hunch, but maybe caused by player changes?

**Comment by skonvols2k** [comment]
Hi,
have a look at this: https://forum.kodi.tv/showthread.php?tid=320547 it's the same problem, the steps to reproduce are well described and there are some pastebin with debug logs. If that isn't sufficient to analyze the problem please let me know, I'll create debugs logs. I'm using Kodi RC3 on Windows 10 x64 and RC2 on Linux x64 (LibreElec) and both are affected. Thanks

**Comment by pkerling** [comment]
"If you want this investigated I recommend to pack a minimal zip file that reliably reproduces the issue."

I see none in the thread.

**Comment by skonvols2k** [comment]
Hi, no problem, can you please point out a guide to do that? Thanks

**Comment by pkerling** [comment]
Not specifically. I think you can find lots of resources on Google on how to create a zip archive out of files.

**Comment by skonvols2k** [comment]
Sorry, of course, I can zip files but which files to you need? :-)

**Comment by pkerling** [comment]
Just as many as are needed to reproduce this problem. From the description I guess 2 videos + 2 subtitle files? But only you will know. That's why we are asking you to supply it.

**Comment by skonvols2k** [comment]
Hi,
I made a video reproducing the problem and showing when it does present itself.
You can found it here: https://drive.google.com/file/d/1KpRzha-Qe1LR887IIT3KkgWjVMYTKAV8/view?usp=sharing (Please let me know when you downloaded it that I will remove ASAP).

I added to Kodi library just 1 video + its subtitle srt file, indexed as "TV Show" then:

- with "Play next video automatically" ON, reproducing the same file:
1. from "TV show" section doesn't load the srt (and kodi crash on stop)
2. from "files" section the srt is loaded (no crash)

- with "Play next video automatically" OFF, reproducing the same file: 
1. from "TV show" section the srt is loaded (no crash)
2. from "files" section the srt is loaded (no crash)

Let me know if need further info.
Thanks

**Comment by pkerling** [comment]
Unfortunately a video of the issue will not help here. We have to reproduce on our side. That is why I ask you to zip files that cause the problem. You can use any dummy video you find online instead of real video files.

**Comment by skonvols2k** [comment]
Hi,
here a sample file: https://drive.google.com/open?id=1FmAhsJ29gpw2KDWq473pJ3gVf8dvqDVK
you should index it as "Tv show" somehow and try to access it from "tv show" library.
Thanks

**Comment by pkerling** [comment]
@FernetMenta Could confirm (Linux)

With a clean Kodi install:
1. Make a folder to put your videos into
1. Create a folder "Star Trek The Next Generation" (or any other show you like :-) there
1. Extract video and sub from https://drive.google.com/open?id=1FmAhsJ29gpw2KDWq473pJ3gVf8dvqDVK to that folder
1. Rename video to `TNG S01E01.mkv` and sub to `TNG S01E01.srt`
1. Add folder from 2) as source to Kodi with content type TV shows, let it scan
1. In main menu select TV shows -> Star Trek [...]
1. Start Episode 1 (important: from the episode listing, not from "recently added" shortcut)
1. Subtitles displayed, yay
1. Stop and go to Settings -> Player, enable "Episodes" in "Play next video automatically"
1. Again in main menu select TV shows -> Star Trek [...] and start episode 1(important: from the episode listing, not from "recently added" shortcut)
1. **Subtitles not displayed, nay**

@skonvols2k These are the instructions I (and @FernetMenta) would have expected from you. Maybe you can learn from the example how to describe things more thoroughly.

**Comment by Dnkhatri** [comment]
I can confirm this on Android as well

**Comment by FernetMenta** [comment]
@pkerling https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L816

make this GetDynPath

**Comment by pkerling** [comment]
@FernetMenta That's it, many thanks! Would have taken me ages to find I guess. Can you give me a sentence on why this has to change so I can put it in the commit message?

**Comment by FernetMenta** [comment]
@pkerling we introduced dynPath because path is the id of a fileitem and must never be changed. In this case path points to the name if the item in videodb and dynPath is set to the actual file. GetDynPath returns dynPath, if set. If there is no dynPath, GetDynPath returns path.
There have been several places where we missed to change GetPath to GetDynPath. THis is one of them.

**Comment by kleczerx** [comment]
With 10K videos on my KODI and years in the IT business, trouble reporting has never been a problem. Yet, your example-layout seems to me as being to absurd. Being concise and direct with information provided to well ‘informed KODI’ experts should not provoke this type of an email.

I will stick to my format, .. Thank you!

RK

**Comment by pkerling** [comment]
@kleczerx Kodi is a huge piece of software with many features and lots of corner-cases. I don't think any developer has in-depth knowledge of all areas. I personally have never used "Play next video automatically" (and didn't even know that option existed). We all have limited time to look into these issues. You never know what tiny detail in e.g. how an item is accessed or what kind of video/subtitle file it is might cause a bug to appear or not. Having all information at hand without having to resort to guesswork is key for not wasting any time here. That's why we ask for reproduction steps to be crystal clear usually and include sample files (of course there are exceptions e.g. when it's immediately obvious what the issue is).

**Comment by pkerling** [comment]
Everyone, pleas test #15116

**Comment by Dnkhatri** [comment]
> Everyone, please test #15116

if it's possible to get an android test build I am willing to test it..

**Comment by enen92** [comment]
@Dnkhatri 

arm: https://jenkins.kodi.tv/job/Android-ARM/12399/artifact/kodi-20181222-2029707b-PR15116-merge-armeabi-v7a.apk

arm64: https://jenkins.kodi.tv/job/Android-ARM64/8563/artifact/kodi-20181222-2029707b-PR15116-merge-arm64-v8a.apk

x86: https://jenkins.kodi.tv/job/Android-X86/4427/artifact/kodi-20181222-2029707b-PR15116-merge-x86.apk

**Comment by skonvols2k** [comment]
Tested on windows, it works!
Thanks guys!

**Comment by FernetMenta** [comment]
> With 10K videos on my KODI and years in the IT business, trouble reporting has never been a problem. Yet, your example-layout seems to me as being to absurd. Being concise and direct with information provided to well ‘informed KODI’ experts should not provoke this type of an email.

> I will stick to my format, .. Thank you!

In other words you were saying that you expect me to invest my time to fix a bug for you and on the other hand you are reluctant to invest yours to make my life little easier.

Good attitude. You are lucky that I didn't read this before I provided the fix.

**Comment by kleczerx** [comment]
If as you say you provided a fix … I thank you, and apologize.

I use two language-sets of subtitles which are beneficial and constantly used by most Kodi people, like myself. I truly apologize for my comments, but you should read your email and refine your approach in bring one’s attention to your concerns.

I will look forward to the next release of Kodi to experience the correction you provided to subtitle operations verses “Play next Video automatically” activation.

While helping to better Kodi, are you aware of my accompanying trouble that I also reported in my “convoluted” format in which when “use video tags” and “Extract video information from files” are activated ‘certain’ newly added videos do not scan to the library’s dbase; once the two options are de-selected these videos do get entered into the Kodi video dbase.

I again am sorry for my statements!

Thank you for your time and experience.

RK

**Comment by kleczerx** [comment]
Martijn,

Thank you for your work.

I have test your sample file “*PR15116-merge-armeabi-v7a.apk”. I general have two subtitles (English and Polish) for any video, and all worked well. The play next video automatically and subtitle loading together work as they should. Thank you for your time and effort.

RK

**Comment by kleczerx** [comment]
Regarding the samples files with corrections made to the problem of subtitle loading and activation of “Play next video automatically”.

Suggest that your corrected Kodi VerRC3 NOT be installed overtop an exiting KODI installation, but only as a fresh install.

As previous stated by me, in email, I tested “* PR15116-merge-armeabi-v7a.apk” which was firstly installed overtop a previous installation (on a “KIII media box”) with approximately 100 video files in the Kodi dbase. At first I actually did have problems so I uninstalled Kodi from the box and then re-install your program as a ‘clean” installation, then all worked well as was required. But, the video library in this case was not extensive and a re-build was quick. Thus, a positive test was confirmed quickly.

I then proceeded to test “*PR15116-merge-arm64-v8a.apk” and again installed over an existing RC3 installation (as was done above in the KIII box) but this time in a Nvidea-Shield box. I was presented with similar subtitle control problems which froze the video followed by Kodi shutting down completely. I followed by deleting the Kodi installation (which had your corrected version installed overtop an existing RC3) AND reinstalled your new file version as a fresh install. All works well now. My video list is extensive and it takes over 24 hours to re-build the dbase.

So, may I advise that if your version is released into the wild you might want to warn users that it be freshly installed and NOT overtop an existing installation.

Unless, you view this as a defect because generally most new versions maybe installed overtop existing earlier version.

RK
